liota/transports/mqtt.py

```diff
@@ -52,21 +54,30 @@ def __init__(self, url, port):
         self.url = url
         self.port = port
         self.client = paho.Client()
+        self.subscribe = self.subscribe
+        self.publish = self.publish
         self.client.on_message = self.on_message
         self.client.on_publish = self.on_publish
         self.client.on_subscribe = self.on_subscribe
         self.connect_soc()
         TransportLayer.__init__(self)
 
     def connect_soc(self):
-        self.client.connect_soc(host=self.url, port=self.port, keepalive=60)
+        self.client.connect(self.url, self.port, 60)
         log.info("Connection Successful")
 
-
     def publish(self, topic, message):
         self.client.publish(topic, message)
-        log.info("Message Sent")
+        log.info("Message {0} sent on topic {1}".format(str(message), str(topic)))
 
     def subscribe(self, topic):
+        log.info("Subscribed to topic {0}".format(str(topic)))
         self.client.subscribe(topic, qos=1)
+        thread.start_new_thread(self.loop_thread, ())
+
+    def loop_thread(self):
         self.client.loop_forever()
+        pass
```

Comment:
Got it. If subscribe happens only once per user app maybe ok for now but what about when every metric potentially is a user app (considering the evolving package design). We have to make sure not to bind a thread to a metric or even to a device.